components/style/values/specified/grid.rs

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use cssparser::{Parser, ToCss};
+use std::fmt;
+use values::NoViewportPercentage;
+use values::computed::ComputedValueAsSpecified;
+
+#[derive(PartialEq, Clone, Debug)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub struct GridLine {
+    pub is_span: bool,
+    pub ident: Option<String>,
+    pub integer: Option<i32>,
+}
+
+impl Default for GridLine {
+    fn default() -> Self {
+        GridLine {
+            is_span: false,
+            ident: None,
+            integer: None,
+        }
+    }
+}
+
+impl ToCss for GridLine {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+        if !self.is_span && self.ident.is_none() && self.integer.is_none() {
+            return dest.write_str("auto")
+        }
+
+        let mut vec = Vec::new();
+        if self.is_span {
+            vec.push("span".to_owned());
+        }
+
+        if let Some(i) = self.integer {
+            vec.push(format!("{}", i));
+        }
+
+        if let Some(ref s) = self.ident {
+            vec.push(s.to_owned());
+        }
+
+        write!(dest, "{}", vec.join(" "))
```

Comment:
Seems like using a vector and allocating for this is overkill, and a single boolean variable should do to get the same effect.